python/cudf/cudf/_lib/io/datasource.pyx

```diff
@@ -4,5 +4,5 @@ from libcpp.memory cimport unique_ptr
 from cudf._lib.cpp.io.types cimport datasource
 
 cdef class Datasource:
-    cdef datasource* get_datasource(self) except +:
-        pass
+    cdef datasource* get_datasource(self) except *:
+        raise Exception("get_datasource() should not be directly invoked here")
```

Comment:
This should be a stronger Exception type, i.e. a `NotImplementedError` or a `ValueError`

@shwina thoughts?

Reply:
Yeah `NotImplementedError` sounds good. Ideally we would make this an `ABC`, but I don't think we can do that for `cdef class`